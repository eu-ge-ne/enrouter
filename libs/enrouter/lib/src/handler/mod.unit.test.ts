import { describe, test, expect } from "vitest";
import * as regexparam from "regexparam";

import { buildRouteHandlers } from "./mod.js";

import type { Route } from "#lib/route/mod.js";

describe("buildRouteHandlers", () => {
  test("from 1 route", () => {
    const route: Route = {
      path: "/",
      test: regexparam.parse("/", true),
      modules: [],
      loaded: false,
    };

    expect(buildRouteHandlers(route)).toMatchSnapshot();
  });

  test("from 2 routes", () => {
    const route: Route = {
      path: "/",
      test: regexparam.parse("/", true),
      modules: [],
      loaded: false,
      tree: [
        {
          path: "/abc",
          test: regexparam.parse("/abc", true),
          modules: [],
          loaded: false,
        },
      ],
    };

    expect(buildRouteHandlers(route)).toMatchSnapshot();
  });
});
