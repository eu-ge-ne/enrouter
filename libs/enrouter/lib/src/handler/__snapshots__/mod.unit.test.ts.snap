// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`buildRouteHandlers > from 1 route 1`] = `
{
  "modules": [],
  "route": {
    "modules": [],
    "path": "/",
  },
  "test": {
    "keys": [],
    "pattern": /\\^\\(\\?=\\$\\|\\\\/\\)/i,
  },
}
`;

exports[`buildRouteHandlers > from 2 routes 1`] = `
{
  "modules": [],
  "route": {
    "modules": [],
    "path": "/",
    "tree": [
      {
        "modules": [],
        "path": "/abc",
      },
    ],
  },
  "test": {
    "keys": [],
    "pattern": /\\^\\(\\?=\\$\\|\\\\/\\)/i,
  },
  "tree": [
    {
      "modules": [],
      "route": {
        "modules": [],
        "path": "/abc",
      },
      "test": {
        "keys": [],
        "pattern": /\\^\\\\/abc\\(\\?=\\$\\|\\\\/\\)/i,
      },
    },
  ],
}
`;
